akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -742,14 +744,18 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
         flushingPromise.future
       }
     implicit val ec = remoteDispatcher
+    flushing.recover { case _ ⇒ Done }.flatMap(_ ⇒ internalShutdown())
+  }
+
+  private def internalShutdown(): Future[Done] = {
+    import system.dispatcher
 
+    killSwitch.abort(ShutdownSignal)
+    topLevelFREvents.loFreq(Transport_KillSwitchPulled, NoMetaData)
     for {
-      _ ← flushing.recover { case _ ⇒ Done }
-      _ = killSwitch.abort(ShutdownSignal)
       _ ← streamsCompleted
+      _ ← taskRunner.stop()
```

Comment:
One more in `handleFatalError`, should there be more cleanup there as well, for example running the entire internal shutdown?